Lib/base64.py

```diff
@@ -601,7 +601,17 @@ def main():
         with open(args[0], 'rb') as f:
             func(f, sys.stdout.buffer)
     else:
-        func(sys.stdin.buffer, sys.stdout.buffer)
+        if sys.stdin.isatty():
+            # gh-gh-138775: read input data at once when reading from stdin
+            # This allows proper handling of EOF (Ctrl+D)
```

Comment:
```suggestion
            # gh-138775: read input data at once when reading from stdin.
```

IIRC, on Windows, the EOF signal is sent through Ctrl+Z, so it's better to not mention this.

Reply:
copy that